src/main/java/org/springframework/integration/kafka/config/xml/KafkaOutboundChannelAdapterParser.java

```diff
@@ -67,6 +67,9 @@ protected AbstractBeanDefinition parseConsumer(final Element element, final Pars
 			kafkaProducerMessageHandlerBuilder.addPropertyValue("partitionIdExpression", partitionIdExpressionDef);
 		}
 
+		IntegrationNamespaceUtils.setValueIfAttributeDefined(kafkaProducerMessageHandlerBuilder, element,
+				"enable-header-routing", "enableHeaderRouting");
```

Comment:
No reason to specify property, if it is can be fabricated from dash-style.
See `IntegrationNamespaceUtils` source code:
```
setValueIfAttributeDefined(builder, element, attributeName,
		Conventions.attributeNameToPropertyName(attributeName), emptyStringAllowed);
```

A `Conventions.attributeNameToPropertyName()` is the trick.